src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -143,6 +200,8 @@ static sd_dhcp_server *dhcp_server_free(sd_dhcp_server *server) {
 
         hashmap_free(server->leases_by_client_id);
 
+        set_free(server->vendor_specific);
```

Comment:
`set_free_with_destructor(server->vendor_specific, vendor_class_specific_free)`

Reply:
I missed we should do this in `network_free`